### PR TITLE
Move download switch away from home

### PR DESCRIPTION
**Pull request by wattsecs**
Move the download switch away from the home fragment. It is now available from the learning dataset selection and from the dataset overview.

### CODE REVIEW

**Review by NicolasVial — COMMENTED** [review]
Everything looks good to me but I got a weird surprise when testing. If i press the downloading button before everything is fully ready on the fragment the app crashes....

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningDatasetSelectionFragment.kt [hunk 1]
```diff
@@ -34,6 +36,13 @@ class LearningDatasetSelectionFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         val dsListFragment =
             childFragmentManager.findFragmentById(R.id.LearningDatasetSelection_dataset_list)
+        
+        if (savedInstanceState == null) {
+            childFragmentManager.commit {
```
Comment (martin-lenweiter) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningDatasetSelectionFragment.kt:43]: Could refactor as this is duplicated with DatasetsOverviewFragment

**Review by martin-lenweiter — CHANGES_REQUESTED** [review]
Looks good, just a minor refactoring should be considered, as well as the issue Nicolas talked about.

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningDatasetSelectionFragment.kt (hunk 1 above)
Comment (wattsecs) [reply app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningDatasetSelectionFragment.kt:43]: I don't see how though tbh

Comment (martin-lenweiter) [reply app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningDatasetSelectionFragment.kt:43]: You could make a util function, but it's only 3 lines so don't bother

**Review by martin-lenweiter — APPROVED** [review]

**Comment by wattsecs** [comment]
> 
> 
> Everything looks good to me but I got a weird surprise when testing. If i press the downloading button before everything is fully ready on the fragment the app crashes....

This should be adressed now

**Comment by Nitwix** [comment]
I don't know if this is hard to fix but I observe the following crash:
- download the database 
- add a new picture in a dataset (from camera)
- click on the category in which you added the picture
- crash

Tell me if you cannot reproduce. If it's hard to fix, it's ok to leave as is, because we don't have a lot of time until the deadline.

E/com.github.HumanLearning2021.HumanLearningApp.model.DefaultImageDisplayer: load failed
    class com.bumptech.glide.load.engine.GlideException: Failed to load resource
    There were 3 root causes:
    java.io.FileNotFoundException(/data/user/0/com.github.HumanLearning2021.HumanLearningApp/cache/936d9f09-4cff-45eb-8264-3210db71c90d: open failed: ENOENT (No such file or directory))
    java.io.FileNotFoundException(open failed: ENOENT (No such file or directory))
    java.io.FileNotFoundException(open failed: ENOENT (No such file or directory))
     call GlideException#logRootCauses(String) for more detail
      Cause (1 of 4): class com.bumptech.glide.load.engine.GlideException: Fetching data failed, class java.io.InputStream, LOCAL
    There was 1 root cause:
    java.io.FileNotFoundException(/data/user/0/com.github.HumanLearning2021.HumanLearningApp/cache/936d9f09-4cff-45eb-8264-3210db71c90d: open failed: ENOENT (No such file or directory))
     call GlideException#logRootCauses(String) for more detail
        Cause (1 of 1): class java.io.FileNotFoundException: /data/user/0/com.github.HumanLearning2021.HumanLearningApp/cache/936d9f09-4cff-45eb-8264-3210db71c90d: open failed: ENOENT (No such file or directory)
      Cause (2 of 4): class com.bumptech.glide.load.engine.GlideException: Fetching data failed, class android.os.ParcelFileDescriptor, LOCAL
    There was 1 root cause:
    java.io.FileNotFoundException(open failed: ENOENT (No such file or directory))
     call GlideException#logRootCauses(String) for more detail
        Cause (1 of 1): class java.io.FileNotFoundException: open failed: ENOENT (No such file or directory)
      Cause (3 of 4): class com.bumptech.glide.load.engine.GlideException: Fetching data failed, class android.content.res.AssetFileDescriptor, LOCAL
    There was 1 root cause:
    java.io.FileNotFoundException(open failed: ENOENT (No such file or directory))
     call GlideException#logRootCauses(String) for more detail
        Cause (1 of 1): class java.io.FileNotFoundException: open failed: ENOENT (No such file or directory)
      Cause (4 of 4): class com.bumptech.glide.load.engine.GlideException: Failed LoadPath{StringUri->Object->Drawable}, LOCAL
        Cause (1 of 2): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{StringUri->Drawable->Drawable}
        Cause (2 of 2): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{StringUri->Bitmap->Drawable}
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.github.HumanLearning2021.HumanLearningApp, PID: 29517
    class com.bumptech.glide.load.engine.GlideException: Failed to load resource
    There were 3 root causes:
    java.io.FileNotFoundException(/data/user/0/com.github.HumanLearning2021.HumanLearningApp/cache/936d9f09-4cff-45eb-8264-3210db71c90d: open failed: ENOENT (No such file or directory))
    java.io.FileNotFoundException(open failed: ENOENT (No such file or directory))
    java.io.FileNotFoundException(open failed: ENOENT (No such file or directory))
     call GlideException#logRootCauses(String) for more detail
      Cause (1 of 4): class com.bumptech.glide.load.engine.GlideException: Fetching data failed, class java.io.InputStream, LOCAL
    There was 1 root cause:
    java.io.FileNotFoundException(/data/user/0/com.github.HumanLearning2021.HumanLearningApp/cache/936d9f09-4cff-45eb-8264-3210db71c90d: open failed: ENOENT (No such file or directory))
     call GlideException#logRootCauses(String) for more detail
        Cause (1 of 1): class java.io.FileNotFoundException: /data/user/0/com.github.HumanLearning2021.HumanLearningApp/cache/936d9f09-4cff-45eb-8264-3210db71c90d: open failed: ENOENT (No such file or directory)
      Cause (2 of 4): class com.bumptech.glide.load.engine.GlideException: Fetching data failed, class android.os.ParcelFileDescriptor, LOCAL
    There was 1 root cause:
    java.io.FileNotFoundException(open failed: ENOENT (No such file or directory))
     call GlideException#logRootCauses(String) for more detail
        Cause (1 of 1): class java.io.FileNotFoundException: open failed: ENOENT (No such file or directory)
      Cause (3 of 4): class com.bumptech.glide.load.engine.GlideException: Fetching data failed, class android.content.res.AssetFileDescriptor, LOCAL
    There was 1 root cause:
    java.io.FileNotFoundException(open failed: ENOENT (No such file or directory))
     call GlideException#logRootCauses(String) for more detail
        Cause (1 of 1): class java.io.FileNotFoundException: open failed: ENOENT (No such file or directory)
      Cause (4 of 4): class com.bumptech.glide.load.engine.GlideException: Failed LoadPath{StringUri->Object->Drawable}, LOCAL
        Cause (1 of 2): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{StringUri->Drawable->Drawable}
        Cause (2 of 2): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{StringUri->Bitmap->Drawable}

**Comment by wattsecs** [comment]
> I don't know if this is hard to fix but I observe the following crash:
> 
>     download the database
>     add a new picture in a dataset (from camera)
>     click on the category in which you added the picture
>     crash
> 
> Tell me if you cannot reproduce. If it's hard to fix, it's ok to leave as is, because we don't have a lot of time until the deadline.
> 
> E/com.github.HumanLearning2021.HumanLearningApp.model.DefaultImageDisplayer: load failed
> class com.bumptech.glide.load.engine.GlideException: Failed to load resource

Thanks, I fixed it.

**Review by Nitwix — APPROVED** [review]
Tested and works as expected! Also, good job on fixing the crash I pointed out.

**Review by bbjubjub2494 — APPROVED** [review]

**Pull request merged**